src/lib/markbind/src/lib/markdown-it/markdown-it-date.js

```diff
@@ -0,0 +1,76 @@
+const Moment = require('moment');
+
+const ERR_MSG = 'Unexpected date format';
+const WEEK_NAN = 'Week is not a number';
+const DAY_INVALID = 'Unexpected day identifier';
+const DAY_OUTOFRANGE = 'Day number should be between 1 - 7 inclusive';
+
+const DAY_TO_MS = 1000 * 60 * 60 * 24;
+const DEFAULT_FORMAT = 'ddd D MMM'; // Thu 27 Aug
+
+const base = Moment('6/27/2019');
+const days = 'mon tue wed thu fri sat sun'.split(' ');
+
+module.exports = require('markdown-it-regexp')(
+  /@date\s?[wW]?(\d+)#(\d|\w{3})(?:\s?\|\s?(\w*))?@/,
+  (match) => {
+    console.log(match);
+    const raw = match[0];
+
+    const {day, week, formatPattern, err} = parseInput(match);
+
+    if (err) return err;
+
+    console.log(day, week, formatPattern);
+
+    return calculateTarget(base, week, day).format(formatPattern);
+  }
+);
+
+/**
+ * Parses and extracts the week, day offset and format from the raw matched expression
+ * @param raw raw matched expression
+ */
+function parseInput(match) {
+  // Parse week number
+  if (isNaN(match[1])) {
+    return showError(WEEK_NAN, raw);
+  }
+  const week = parseInt(match[1]);
+
+  // Parse day
+  let day = match[2].toLowerCase();
+  if (isNaN(day)) {
```

Comment:
Should check the length instead, 3 character / 1 character. `\w{3}` also matches 3 digits. which may not be intended.

Reply:
If day is `001` it'll still be correct and interpreted as `1`. Is this intended? Probably not a big deal either way